secret-engine/src/main/java/com/sequenceiq/cloudbreak/service/secret/service/SecretService.java

```diff
@@ -66,22 +65,19 @@ public void init() {
      */
     public String put(String key, String value) throws Exception {
         long start = System.currentTimeMillis();
-        boolean exists = vaultRetryService.tryReadingVault(() -> persistentEngine.exists(key));
-        long duration = System.currentTimeMillis() - start;
-        metricService.gauge(MetricType.VAULT_READ, duration);
-        LOGGER.trace("Secret read took {} ms", duration);
-        if (exists) {
-            throw new InvalidKeyException(format("Key: %s already exists!", key));
-        }
-        start = System.currentTimeMillis();
         String secret = vaultRetryService.tryWritingVault(() -> persistentEngine.put(key, value));
-        duration = System.currentTimeMillis() - start;
+        long duration = System.currentTimeMillis() - start;
         metricService.gauge(MetricType.VAULT_WRITE, duration);
         LOGGER.trace("Secret write took {} ms", duration);
         metricService.incrementMetricCounter(() -> "secret.write." + convertSecretToMetric(secret));
         return secret;
     }
 
+    public String update(String secret, String newValue) throws Exception {
```

Comment:
Would this automatically create a v2 version of the secret?

Reply:
yes

Reply:
sure, good idea